src/lib/asyncReader.ts

```diff
@@ -0,0 +1,10 @@
+let AsyncReader;
```

Comment:
really dont think this should be its own file, let alone its own class

Reply:
Standard OO practice is to make a class out of every I/O operation. Especially important in this case since it's kind of general logic that nothing else needs to know about, and it also "exports" it's own "class" (Promise<FileReader>)